tests/unit/lms/resources/_js_config_test.py

```diff
@@ -27,6 +27,106 @@ def test_enable_content_item_selection_mode(self, js_config):
         assert js_config.config["mode"] == "content-item-selection"
 
 
+class TestAddCanvasFileID:
+    """Unit tests for JSConfig.add_canvas_file_id()."""
+
+    def test_it_adds_the_via_url_callback_url(self, js_config):
+        js_config.add_canvas_file_id("example_canvas_file_id")
+
+        assert (
+            js_config.config["urls"]["via_url_callback"]
+            == "http://example.com/api/canvas/files/example_canvas_file_id/via_url"
+        )
+
+    def test_it_sets_the_canvas_file_id(self, js_config):
+        js_config.add_canvas_file_id("example_canvas_file_id")
+
+        assert (
+            js_config.config["submissionParams"]["canvas_file_id"]
+            == "example_canvas_file_id"
+        )
+
+
+class TestAddDocumentURL:
+    """Unit tests for JSConfig.add_document_url()."""
+
+    def test_it_adds_the_via_url(self, js_config):
+        js_config.add_document_url("example_document_url")
+
+        assert (
+            js_config.config["urls"]["via_url"]
+            == "http://TEST_VIA_SERVER.is/example_document_url?via.open_sidebar=1&via.request_config_from_frame=http%3A%2F%2Fexample.com"
+        )
```

Comment:
Ah, yeah I forgot to add a patch after adding the via_url import. I've added a via_url patch fixture, so the test can now just do:

```python
assert js_config.config["urls"]["via_url"] == via_url.return_value
```

Since we're not actually interested in any part of the via_url string at all (that's already tested elsewhere in `via_url()`'s tests), `JSConfig`'s job is just to call `via_url` and inject its result into the config